pkg/apis/eventing/v1alpha1/trigger_conversion.go

```diff
@@ -34,14 +34,19 @@ func (source *Trigger) ConvertTo(ctx context.Context, obj apis.Convertible) erro
 		sink.Spec.Subscriber = source.Spec.Subscriber
 		if source.Spec.Filter != nil {
 			sink.Spec.Filter = &v1beta1.TriggerFilter{
-				Attributes: make(v1beta1.TriggerFilterAttributes, 0),
 			}
 			if source.Spec.Filter.Attributes != nil {
+				sink.Spec.Filter = &v1beta1.TriggerFilter{
+					Attributes: make(v1beta1.TriggerFilterAttributes, 0),
+				}
 				for k, v := range *source.Spec.Filter.Attributes {
 					sink.Spec.Filter.Attributes[k] = v
 				}
 			}
 			if source.Spec.Filter.DeprecatedSourceAndType != nil {
+				sink.Spec.Filter = &v1beta1.TriggerFilter{
+					Attributes: make(v1beta1.TriggerFilterAttributes, 0),
```

Comment:
could we not just make this slice at least two already given that we default to adding a `source` and `type` attribute?